### PR TITLE
message list [nfc]: Small cleanups around BackgroundData to prepare for adding more.

### PR DESCRIPTION
**Pull request by gnprice**
Inspired by looking back at this code while studying #3811.

@agrawal-d, after this PR you should be able to get the `is_admin` bit by just picking out `ownUser` and saying `ownUser.is_admin`, without any additional plumbing at the layers above the spot where you actually need it.

### CODE REVIEW

File: src/webview/MessageList.js [hunk 1]
```diff
@@ -334,15 +334,15 @@ export default connect<SelectorProps, _, _>((state, props: OuterProps) => {
...
     auth: getAuth(state),
     debug: getDebug(state),
     flags: getFlags(state),
     mute: getMute(state),
     ownEmail: getOwnEmail(state),
     ownUserId: getOwnUser(state).user_id,
-    allImageEmojiById: getAllImageEmojiById(state),
-    subscriptions: getSubscriptions(state),
     twentyFourHourTime: getRealm(state).twentyFourHourTime,
+    subscriptions: getSubscriptions(state),
```
Comment (ray-kraesig) [inline src/webview/MessageList.js]: `"t" < "s"`?

File: src/webview/MessageList.js [hunk 2]
```diff
@@ -72,6 +72,7 @@ export type BackgroundData = $ReadOnly<{
   flags: FlagsState,
   mute: MuteState,
   ownUser: User,
+  theme: ThemeName,
```
Comment (ray-kraesig) [inline src/webview/MessageList.js]: I'm not sure what jsdoc you're referring to when you say, in the commit message,

> as explained in the jsdoc on the type.

The jsdoc on `BackgroundData` in particular states: `Data about the user, the realm, and all known messages.`, which doesn't seem consonant with including `theme`.

File: src/webview/MessageList.js [hunk 3]
```diff
@@ -78,7 +82,12 @@ export type BackgroundData = $ReadOnly<{
 }>;
 
 type SelectorProps = {|
+  // Data about the user, the realm, and all known messages.
```
Comment (ray-kraesig) [inline src/webview/MessageList.js]: This seems likely to diverge confusingly from its source.

File: src/webview/MessageList.js (hunk 1 above)
Comment (gnprice) [reply src/webview/MessageList.js]: Ha, thanks for the catch. The two lists disagreed, but then I settled it the wrong way :)

File: src/webview/MessageList.js (hunk 2 above)
Comment (gnprice) [reply src/webview/MessageList.js]: The theme is a global choice made by the user, so I think it fits broadly under "about the user".

The main reason I think it belongs here is the other side of the distinction, in the next paragraph: it's independent of the specific narrow and specific messages we're displaying.

Which also goes along with being suitable to include in the bundle that's wired through to a diverse set of helpers below: many of those helpers deal with a particular message, and when they also get data about the current narrow, that creates a risk of confusion (just like in #3811) about which information applies.

File: src/webview/MessageList.js (hunk 3 above)
Comment (gnprice) [reply src/webview/MessageList.js]: Edited to be in terms of the other side of the distinction. (Though in practice I think this description is likely to be pretty stable too.)

**Comment by gnprice** [comment]
Thanks for the review! Pushed a revised version.

**Pull request merged**